Tella/Scenes/Audio/AudioPlayer/Views/EditAudioView.swift

```diff
@@ -0,0 +1,255 @@
+                }) {
+                    Image("edit.audio.cut")
+                        .resizable()
+                        .frame(width: 24, height: 24)
+                }
+            }
+            
+        }
+        .frame(height: 30)
+        .padding(16)
+    }
+    
+    var timeLabelsView: some View {
+        HStack(spacing: 0) {
+            ForEach(editAudioViewModel.generateTimeLabels(), id: \.self) { time in
+                Text(time)
+                    .font(.system(size: 12, weight: .regular))
+                    .foregroundColor(.white.opacity(0.32))
+                if time != editAudioViewModel.generateTimeLabels().last {
+                    Spacer()
+                }
+            }
+        }.frame(width: kTrimViewWidth)
+            .padding([.top], 70)
+    }
+    
+    var displayTimeLabels: some View {
+        VStack {
+            Text(editAudioViewModel.currentTime)
+                .font(.custom(Styles.Fonts.regularFontName, size: 50))
+                .foregroundColor(.white)
+            Text(editAudioViewModel.audioPlayerViewModel.duration)
+                .font(.custom(Styles.Fonts.regularFontName, size: 14))
+                .foregroundColor(.gray)
+        }  .padding(.top, 70)
+    }
+    
+    
+    var controlButtonsView: some View {
+        HStack(spacing: 64) {
+            Button(action: { self.undo() }) {
+                Image("cancel.edit.file")
+                    .resizable()
+                    .frame(width: 50, height: 50)
+                    .foregroundColor(.gray)
+            }
+            
+            Button(action: { editAudioViewModel.handlePlayButton() }) {
+                Image(editAudioViewModel.playButtonImageName)
+                    .resizable()
+                    .frame(width: 50, height: 50)
+                    .foregroundColor(.gray)
+            }
+            
+        }
+        .padding(.top, 64)
+    }
+    
+    private func undo() {
+        editAudioViewModel.startTime = 0.0
+        editAudioViewModel.endTime = editAudioViewModel.timeDuration
+        leadingGestureValue = 0.0
+        trailingGestureValue = kTrimViewWidth
+    }
+    private func closeView() {
+        editAudioViewModel.isPlaying = false
+        if editAudioViewModel.isDurationHasChanged() {
+            isBottomSheetShown = true
+        }else  {
+            sheetManager.hide()
+            isPresented = false
+        }
+    }
+    
+    private func handleTrimState(value:ViewModelState<Bool>) {
+        switch value {
+        case .loaded(let isSaved):
+            if isSaved {
+                self.isPresented = false
+                self.sheetManager.hide()
+                Toast.displayToast(message: LocalizableVault.editFileSavedToast.localized)
+            }
+        case .error(let message):
+            Toast.displayToast(message: message)
+        default:
+            break
+        }
+    }
+
+}
+
+
+
+private struct TrimAudioSliderView: View {
```

Comment:
I think it's better to move this view to an other class

Reply:
I think no need to move it  as it is only specific to the edit audioView I mentioned it as a private class